### DISCUSSION TITLE
suivi d'achats - propositions features à discuter

### DISCUSSION THREAD

**Discussion by JenniferStephan** [betagouv/ma-cantine#1024] [Idées]
- [ ] commencer les stats et calcul par création de champs https://github.com/betagouv/ma-cantine/issues/1023#issue-1126001768

- [x] permettre la création massive de lignes par import de fichier

**Comment by alemangui** [comment]
On a aussi quelques propositions d'Helen [ici](https://github.com/betagouv/ma-cantine/discussions/995#discussioncomment-2111013), notamment l'ajout d'un champ _description_ pour savoir ce qu'on a rentré. C'est assez simple à mettre en place et ça peut dépanner.

**Comment by alemangui** [comment]
Pas sur de bien saisir le renseignement de la valeur totale de l'année en cours. Elle change assez souvent non ? On s'attend à que l'utilisateur le mette à jour régulièrement ? Si oui, qu'est-ce qu'iel en tire comme bénéfice ?

**Comment by Greta712** [comment]
créer un bouton supprimer après la validation d'une ligne d'achat

**Reply by alemangui** [reply]
Tout à fait, c'est prévu ici : https://github.com/betagouv/ma-cantine/issues/923

**Reply by hfroot** [reply]
D'ailleurs @Greta712 Alejandro à mis ça en staging si tu veux le tester :)

**Comment by valeriemerle** [comment]
A t-on dejà une idée des utilisateurs qui utilisent l'outil de suivi des achats? A -ton des fichiers exportables?
En prévision du GT suivi du CNRC du 17/02, peut on proposer de reprendre les "tableaux de suivi (famille/catégorie) qui seront choisi? Pour le moment il y a qq différences dans les familles de produits et les catégories.

**Reply by hfroot** [reply]
On a une discussion sur les export de données d'achats ici : https://github.com/betagouv/ma-cantine/discussions/989 :)

**Reply by hfroot** [reply]
@valeriemerle C'est a dire quoi "reprendre" dans ce contexte ?

**Comment by valeriemerle** [comment]
Sur l'outil de suivi qq remarques sur le wording et le contenu (désolée ça m'avait échappé avant) :
Les "catégories" sont en fait des familles de produits (viandes, volailles...). Le GT suivi a retenu 14 familles pour le moment correspondant à des lots classiques pour les acheteurs. Cette nomenclature sera validée sous peu.
les "caractéristiques" sont des catégories
Dans les caractéristiques : il n'y a pas besoin de séparer  "bio" et "en conversion bio" la dernière catégorie n'est souvent pas indiqué sur les produits. Peche durable est à remplacer par "Ecolabel pêche durable pour éviter tout problème de confusion avec des labels privés comme MSC qui ne sont pris en compte dans cette catégorie.
Le GT suivi a fait le choix de ne pas laisser "apparent" la catégorie "produit équivalent.." .

**Comment by paulineeyherabide** [comment]
Possibilité de cumuler plusieurs catégories sur une ligne d'achat (compliqué de détailler pour les factures de grossistes cumulant différentes catégories) avec le détail des prix HT par catégorie et le global.

**Reply by hfroot** [reply]
@paulineeyherabide on imagine que les utilisateurs vont créer une ligne par chaque article dans un facture (e.g. "œufs" sont un achat, "bœuf" un autre). Bientôt on va ajouter un champ pour une description d'achat pour clarifier ça.
1. Comme ça, est-ce que tu vois des cas où plusieurs catégories seront nécessaires ?
2. Est-ce que tu penses que cette méthode d'utilisation va convenir les utilisatrices ?

**Reply by paulineeyherabide** [reply]
@hfroot merci :)

1. C'est un utilisateur qui m'a transmis cette info car dans une même facture, il peut y avoir plusieurs achats. En gros, il voudrait pouvoir ajouter tous les achats d'une même facture sur une même ligne, avec un détail des prix par achat par catégorie et un global. Mais ça impliquerait une ligne = une facture et ce 'est pas l'idée je pense. Je ne sais pas si je suis claire ❌ 
2. J'ai eu un seul retour à ce sujet, donc peut-être attendre de voir si on a d'autres retour en ce sens..

**Reply by hfroot** [reply]
@paulineeyherabide merci pour les précisions, je vois pourquoi l'utilisateur demande ça. Peut-être on peut penser d'un façon de rendre plus facile l'ajout de plusieurs achats pour le même facture.
Est-ce que un facture = un fournisseur ou pas forcement ?

**Reply by paulineeyherabide** [reply]
@hfroot Je dirais que oui

**Comment by Greta712** [comment]
j'ai eu un retour d'une utilisatrice sur l'usage "suivi des achats" qu'elle trouve limité. Elle suggère les fonctions suivantes:
-une fonction tri
-une fonction impression
-des sous-classes
la saisie de données brutes sans pouvoir l'utiliser hors du site ma cantine ne présente pas beaucoup d'intérêt pour elle

**Reply by Greta712** [reply]
J'ai reçu une remarque à nouveau sur le tableau d'export, peut-il être similaire à celui du tableau suivi des appros?

**Comment by gauthierandre** [comment]
Possibilité de saisie automatique : ex fournisseur. 
Un retour utilisateur sur le parcours 
"ne peut-on pas quelque chose de totalement automatisé" : ex suggestion de type de produit, fournisseur généré à partir des saisies précédentes etc.

**Comment by ghost** [comment]
Fonctionnalité "Import", qui permettrait d'avoir la création automatique des lignes de factures suite à l'import d'un fichier de suivi d'achats, pour les gestionnaires qui souhaiteraient passer en cours d'année d'un tableur Excel à Mes Achats sans avoir à re-saisir toutes les lignes de factures déjà renseignées sur le tableur. Besoin exprimé par un utilisateur (H. F., le 30/03).

**Comment by ghost** [comment]
Une autre fonctionnalité demandée par le même utilisateur : dans le cas d'une cuisine centrale, pouvoir ventiler un achat sur plusieurs cantines satellites en sélectionnant les X cantines satellites concernées par l'achat et en indiquant, pour chacune d'entre elles, à quelle proportion.
